test/e2e/integreatly_test.go

```diff
@@ -0,0 +1,141 @@
+	operator "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	retryInterval                    = time.Second * 5
+	timeout                          = time.Second * 60
+	deploymentRetryInterval          = time.Second * 30
+	deploymentTimeout                = time.Minute * 20
+	cleanupRetryInterval             = time.Second * 1
+	cleanupTimeout                   = time.Second * 5
+	installationCleanupRetryInterval = time.Second * 20
+	installationCleanupTimeout       = time.Minute * 2 //Longer timeout required to allow for finalizers to execute
+	intlyNamespacePrefix             = "intly-"
+)
+
+func TestIntegreatly(t *testing.T) {
+
+	logf.SetLogger(logf.ZapLogger(true))
+
+	installationList := &operator.InstallationList{}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, installationList)
+	if err != nil {
+		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+	}
+	// run subtests
+	t.Run("integreatly", func(t *testing.T) {
+		t.Run("Cluster", IntegreatlyCluster)
+	})
+
+}
+
+func waitForProductDeployment(t *testing.T, f *framework.Framework, ctx *framework.TestCtx, product, deploymentName string) error {
+	namespace := intlyNamespacePrefix + product
+	t.Logf("Checking %s:%s", namespace, deploymentName)
+
+	start := time.Now()
+	err := e2eutil.WaitForDeployment(t, f.KubeClient, namespace, deploymentName, 1, deploymentRetryInterval, deploymentTimeout)
+	if err != nil {
+		return err
+	}
+
+	end := time.Now()
+	elapsed := end.Sub(start)
+
+	t.Logf("%s:%s up, waited %d", namespace, deploymentName, elapsed)
+	return nil
+}
+
+func integreatlyWorkshopTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("could not get namespace: %deploymentName", err)
+	}
+
+	// create installation custom resource
+	workshopInstallation := &operator.Installation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "e2e-workshop-installation",
+			Namespace: namespace,
+		},
+		Spec: operator.InstallationSpec{
+			Type:             "workshop",
+			NamespacePrefix:  intlyNamespacePrefix,
+			RoutingSubdomain: "apps.example.com",
+			MasterURL:        "http://console.apps.example.com",
+			SelfSignedCerts:  true,
+		},
+	}
+	// use TestCtx's create helper to create the object and add a cleanup function for the new object
+	err = f.Client.Create(goctx.TODO(), workshopInstallation, &framework.CleanupOptions{TestContext: ctx, Timeout: installationCleanupTimeout, RetryInterval: installationCleanupRetryInterval})
+	if err != nil {
+		return err
+	}
+
+	//Auth Stage - verify operators deploy
+	err = waitForProductDeployment(t, f, ctx, "rhsso", "keycloak-operator")
```

Comment:
The "rhsso", "keycloak-operator" params refer to namespace's name? If so there could be the problem with the namespace prefix.

Reply:
waitForProductDeployment adds the prefix here https://github.com/integr8ly/integreatly-operator/pull/128/files/8ce92eda791272eb72fa63dbb8f875648ee01056#diff-e3e74a4fcd6e24a416beb0a2b8f02762R48, and is set in the test here https://github.com/integr8ly/integreatly-operator/pull/128/files/8ce92eda791272eb72fa63dbb8f875648ee01056#diff-e3e74a4fcd6e24a416beb0a2b8f02762R28

Reply:
OK I see. Sorry I didn't see it.